src/Features/Core/Portable/CodeRefactorings/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,198 @@
+            where TExpressionSyntax : SyntaxNode
+            where TIfStatementSyntax : SyntaxNode
+        {
+            protected readonly ISyntaxFactsService _syntaxFacts;
+            protected readonly SemanticModel _semanticModel;
+            private TExpressionSyntax _switchExpression;
+
+            public Analyzer(ISyntaxFactsService syntaxFacts, SemanticModel semanticModel)
+            {
+                _syntaxFacts = syntaxFacts;
+                _semanticModel = semanticModel;
+            }
+
+            public async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+            {
+                var document = context.Document;
+                var cancellationToken = context.CancellationToken;
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+                var ifStatement = root.FindNode(context.Span).FirstAncestorOrSelf<TIfStatementSyntax>();
+                if (ifStatement == null)
+                {
+                    return;
+                }
+
+                if (ifStatement.ContainsDiagnostics)
+                {
+                    return;
+                }
+
+                if (!ifStatement.GetFirstToken().GetLocation().SourceSpan.IntersectsWith(context.Span))
+                {
+                    return;
+                }
+
+                if (!CanConvertIfToSwitch(ifStatement))
+                {
+                    return;
+                }
+
+                var switchDefaultBody = default(TStatementSyntax);
+                var switchSections = new List<(IEnumerable<IPattern>, TStatementSyntax)>();
+
+                // Iterate over if-else statement chain.
+                foreach (var (body, condition) in GetIfElseStatementChain(ifStatement))
+                {
+                    if (condition == null)
+                    {
+                        switchDefaultBody = body;
+                        break;
+                    }
+
+                    var operands = GetLogicalOrExpressionOperands(condition);
+                    var patterns = new List<IPattern>();
+
+                    // Iterate over "||" operands to make a case label per each condition.
+                    foreach (var operand in operands.Reverse())
+                    {
+                        var pattern = CreatePatternFromExpression(operand);
+                        if (pattern == null)
+                        {
+                            return;
+                        }
+
+                        patterns.Add(pattern);
+                    }
+
+                    switchSections.Add((patterns, body));
+                }
+
+                context.RegisterRefactoring(new MyCodeAction(Title, c =>
+                    UpdateDocumentAsync(root, document, ifStatement, switchDefaultBody, switchSections)));
+            }
+
+            protected bool IsEquivalentToSwitchExpression(TExpressionSyntax expression)
```

Comment:
I would like this renamed.  Even if it's verbose, somethign like SetInitialExpressionOrCheckThatExpressionIsEquivalent